conjur/wrapper/http_wrapper.py

```diff
@@ -93,6 +89,24 @@ def invoke_endpoint(http_verb, endpoint, params, *args, check_errors=True,
 
     return response
 
+def invoke_request(request_method, url, *args, query, ssl_verify, auth, headers):
+    """
+    This method preforms the actual request and catches possible SSLErrors to
+    perform more user-friendly messages
+    """
+    try:
+        return request_method(url, *args,
+                              params=query,
+                              verify=ssl_verify,
+                              auth=auth,
+                              headers=headers)
+
+    except requests.exceptions.SSLError as ssl_error:
+        host_mismatch_message = re.search("(?:doesn't match either of)", str(ssl_error))
+        if host_mismatch_message:
+            raise CertificateHostnameMismatchException from ssl_error
+        raise ssl_error
+
```

Comment:
I think this is best suited to create request_method here and pass http_verb as a parameter. `request_method = getattr(requests, http_verb.name.lower())`
less buggy for my opinion